### PR TITLE
[IMP] product_multi_ean: Remove check EAN restrictions

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa
Close other PR: https://github.com/OCA/stock-logistics-barcode/pull/205
Only remove Ean check restriction

### CODE REVIEW

File: product_multi_ean/models/product_product.py [hunk 1]
```diff
@@ -28,18 +28,6 @@ class ProductEan13(models.Model):
         required=True,
     )
 
-    @api.multi
-    @api.constrains('name')
-    @api.onchange('name')
-    def _check_name(self):
-        barcode_obj = self.env['barcode.nomenclature']
-        for record in self.filtered('name'):
-                if not barcode_obj.check_ean(record.name):
```
Comment (rousseldenis) [inline product_multi_ean/models/product_product.py:37]: In fact, the function called in this check was misused.

You cannot just remove this but change the logic.

You have to call this :

https://github.com/odoo/odoo/blob/11.0/addons/barcodes/models/barcodes.py#L125

And then, call :

https://github.com/odoo/odoo/blob/11.0/addons/barcodes/models/barcodes.py#L57

with the first provided encoding.

If you want to implement a new one (GS1), simply override these functions

**Review by rousseldenis — DISMISSED** [review]
IMHO, the need is to change the implementation as it was not correct before.

The removal of this check is not needed

File: product_multi_ean/models/product_product.py (hunk 1 above)
Comment (sergio-teruel) [reply product_multi_ean/models/product_product.py:37]: By default, Odoo allow add any barcode to product barcode field, ¿Why this module add this restriction?

Comment (sergio-teruel) [reply product_multi_ean/models/product_product.py:37]: I Know that this restrictions are used when a barcode is readed... but not when a product is created

Comment (rousseldenis) [reply product_multi_ean/models/product_product.py:37]: You have to ask the author :smile: 

See refactoring commit :

https://github.com/OCA/stock-logistics-barcode/commit/3fa99ac3a558d1913f72fbe5549346fd0fd3d8cb#diff-71824b7fc2c784066fee1144d32b283a

@pedrobaeza

Comment (pedrobaeza) [reply product_multi_ean/models/product_product.py:37]: That made sense when done on previous version, but not now. It's something that should be taken into account when migrating

**Review by pedrobaeza — APPROVED** [review]

**Comment by rousseldenis** [comment]
Maybe is it too much change on this stable version as users used to this module will wait for constrain triggering (and maybe after update will not pay attention to errors that do not appear anymore...). IMHO ok for v12

**Comment by pedrobaeza** [comment]
@rousseldenis this module was merged not too much time ago, and the problem is not the behavior. It's that this breaks any other module/behavior adding other type of barcodes, so this is simply broken and must be fixed, no matter change of behavior.

**Comment by rousseldenis** [comment]
That's why I proposed to change constraint implementation to make it work for any type of barcode.

I would agree for a constraint deletion only if you implement another module for that purpose (that will allow multi ean with/without constraint).

Or with an option.

**Comment by pedrobaeza** [comment]
Odoo had a constraint on previous versions, and this module mimmicks the same for multi-EAN. Now there's no constraint on Odoo, so this module mustn't have as well. Please don't complicate this break-fixing PR.

**Comment by rousseldenis** [comment]
What I said is that you are not the only users of this module and I try to not change the behaviour for existing users.

But as I'm not part of these users, do what you think.

**Comment by pedrobaeza** [comment]
Thanks for your trust, @rousseldenis I can assure you that the feature is misused here for the main barcode. @sergio-teruel any way, what about providing an alternative removing `barcode` as computed and providing other field for that?

**Comment by sergio-teruel** [comment]
We can create a field "ean13" (like odoo had in older versions) but I think that user can be confuse if see two fields for barcodes in product view...

**Comment by pedrobaeza** [comment]
What do you think, @rousseldenis ?

**Comment by Joaco1980** [comment]
I do not find the problem of removing a restriction that the barcode field does not have without installing the multi ean module. And if you put a selector in inventory settings if you want to validate bar codes like ean13.
There are many types of barcode coding, why put limitations on only one type of coding?

**Comment by pedrobaeza** [comment]
@Joaco1980 please approve the PR then using GitHub mechanism.

**Review by Joaco1980 — APPROVED** [review]

**Pull request merged**